src/utils/cluster_utils.cpp

```diff
@@ -14,37 +15,42 @@ Eigen::VectorXd bayesmix::cluster_estimate(
   unsigned int n_data = alloc_chain.cols();
   Eigen::MatrixXd mean_diss = Eigen::MatrixXd::Zero(n_data, n_data);
   std::vector<Eigen::SparseMatrix<double> > all_diss;
+  progresscpp::ProgressBar bar(n_iter, 60);
+  std::cout << "Computing mean dissimilarity... " << std::flush;
 
-  // Loop over iterations
-  for (size_t i = 0; i < n_iter; i++) {
-    // Find and all nonzero entries of the dissimilarity matrix
-    std::vector<Eigen::Triplet<double> > triplets_list;
-    triplets_list.reserve(n_data * n_data / 4);
-    for (size_t j = 0; j < n_data; i++) {
-      for (size_t k = 0; k < j; k++) {
-        if (alloc_chain(i, j) == alloc_chain(i, k)) {
-          triplets_list.push_back(Eigen::Triplet<double>(j, k, 1.0));
+  // Loop over pairs (i,j) of data points
+  for (int i = 0; i < n_data; i++) {
+    for (int j = 0; j < i; j++) {
+      // Compute mean dissimilarity for (i,j) by looping over iterations
+      for (int k = 0; k < n_iter; k++) {
```

Comment:
I would replace this inner loop with
```
mean_diss(i, j) = 1.0 * ((alloc_chain.col(i) - alloc_chain.col(j)) == 0).count() / n_iter;
```
this should trigger AVX vectorization and cache optimization from Eigen

Reply:
Ooooh that's what you meant. Sorry, I really don't know that stuff like this even exists in C++, and thought these optimizations were being used automatically

Reply:
It doesn't work though, due to mismatching types in the comparison. Should I replace the 0 with a vector of zeros?

Reply:
I thought it would work since it's very similar to an example in the Eigen manual: see Boolean reductions here : https://eigen.tuxfamily.org/dox/group__TutorialReductionsVisitorsBroadcasting.html

Maybe explicitly create a temporary with the difference?

Reply:
Here! The problem was that it only works with Arrays, so an appropriate temp works